packages/nextjs-mf/src/loaders/share-scope-hoist.ts

```diff
@@ -1,71 +1,76 @@
+}
 
 /**
- *
  * Requires either the default delegate module or a custom one
- *
  */
 export default function patchDefaultSharedLoader(
   this: LoaderContext<Record<string, unknown>>,
   content: string
 ) {
-  const { shared } = this.getOptions() as Record<string, string>;
-
   if (content.includes('hasShareHoist')) {
     return content;
   }
-  let shareScopes = '';
-  let eagerShared = {
-    scope: '',
-    sideload: '',
-  };
-  if (shared) {
-    // @ts-ignore
-    eagerShared = Object.entries(shared).reduce(
-      (acc, sharedPackage) => {
-        const name = sharedPackage[0];
-        const params = sharedPackage[1];
 
-        // @ts-ignore
-        if (params.eager === true) {
-          //@ts-ignore
-          acc.scope += `
-       ${JSON.stringify(name)}: {
-         "0": {
-          eager: true,
-          loaded: true,
-          get: () => () => require(${JSON.stringify("!!" + name + "?pop")}),
-        }
-        },`;
-          acc.sideload += `
-      __webpack_modules__[require.resolveWeak(${JSON.stringify(
-        name
-      )})] = __webpack_modules__[require.resolveWeak(${JSON.stringify(
-            '!!' + name + '?pop'
-          )})];
-      `;
-        }
-        return acc;
-      },
-      { scope: '', sideload: '' }
-    );
+  const { shared } = this.getOptions() as Record<string, any>;
+  const eagerShared: EagerShared = buildEagerShared(shared);
 
-    shareScopes =
-      `
-  const eager = {` +
-      eagerShared.scope +
-      `};
-  __webpack_share_scopes__.default = __webpack_share_scopes__.default || {};
-  Object.assign(__webpack_share_scopes__.default || {}, eager);
-  `;
-  }
+  const shareScopes = buildShareScopes(eagerShared.scope);
 
-  return [
+  const result = [
     "console.log('initializing internal-module-hoist for:', __webpack_runtime_id__)",
     shareScopes,
     eagerShared.sideload,
     '//hasShareHoist',
     content,
   ].join('\n');
+
+  // Cache the generated result
+  if (this.cacheable) {
+    this.cacheable();
+  }
+
+  return result;
+}
+
+function buildEagerShared(shared: Record<string, any>): EagerShared {
+  if (!shared) {
+    return { scope: '', sideload: '' };
+  }
+
+  return Object.entries(shared).reduce(
+    (acc: EagerShared, [name, params]) => {
+      if (params.eager === true) {
+        acc.scope += `
+        ${JSON.stringify(name)}: {
+          "0": {
+            eager: true,
+            loaded: true,
+            get: () => () => require(${JSON.stringify('!!' + name + '?pop')}),
+          }
+        },`;
+        acc.sideload += `
+        __webpack_modules__[require.resolveWeak(${JSON.stringify(
+          name
+        )})] = __webpack_modules__[require.resolveWeak(${JSON.stringify(
+          '!!' + name + '?pop'
+        )})];
+        `;
+      }
+      return acc;
+    },
+    { scope: '', sideload: '' }
+  );
+}
+
+function buildShareScopes(eagerSharedScope: string): string {
+  return `
+  const eager = {${eagerSharedScope}};
```

Comment:
shouldn't we stringify this? 
```suggestion
  const eager = {${JSON.stringify(eagerSharedScope)}};
```

Reply:
test if it builds still, technically its already a string, eager share scope doesn't have wrapping braces so i don think it can be stringified again since its not an object